app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/CollectionViewHolder.kt

```diff
@@ -92,47 +93,47 @@ class CollectionViewHolder(
     }
 
     private fun updateCollectionUI() {
-        view.collection_title.text = collection.title
-
-        val hostNameList = collection.tabs.map { it.url.urlToTrimmedHost().capitalize() }
-
-        var tabsDisplayed = 0
-        val tabTitlesList = hostNameList.joinToString(", ") {
-            if (it.length > maxTitleLength) {
-                it.substring(
-                    0,
-                    maxTitleLength
-                ) + "..."
+        launch(Dispatchers.Main) {
+            view.collection_title.text = collection.title
+
+            val collectionDescription = collection.tabs
+                .map { it.url.urlToTrimmedHost(view.context).capitalize() }
+                .map {
+                    if (it.length > maxTitleLength) {
+                        it.substring(
+                            0,
+                            maxTitleLength
+                        ) + "…"
+                    } else {
+                        it
+                    }
+                }.joinToString(", ")
```

Comment:
This code and [this code][1] is identical.

I'd put this in an override function for 

```kotlin
fun Collection<Tab>.description(): String {

}
```
so you can use it in two places.

Either do it now ✅, or file an issue for it ✅.

[1]: https://github.com/mozilla-mobile/fenix/pull/3191/files#diff-ae7a7f880e17ab61958437bea4fd1e89R88

Reply:
Done! 😄